deblur/workflow.py

```diff
@@ -382,40 +347,122 @@ def write_biom_table(table, biom_fp):
-    """Merge multiple BIOM tables into one.
+    Returns
+    -------
+    names : list of tuples of (string,string)
+        list of tuples of:
+            name of fasta files to be added to the biom table
+            sampleid (file names without the file_end and path)
+    """
+    logger = logging.getLogger(__name__)
+    logger.debug('get_files_for_table input dir %s, '
+                 'file-ending %s' % (input_dir, file_end))
 
-    Code taken from QIIME's merge_otu_tables.py
+    names = []
+    # scan the input directory entries
+    for cfile in listdir(input_dir):
+        cname = join(input_dir, cfile)
+        # we want only files
+        if not isfile(cname):
+            continue
+        # test if the filename has the ending we're expecting
+        if cfile.endswith(file_end):
+            sampleid = cfile[:-len(file_end)]
+            names.append((cname, sampleid))
+
+    logger.debug('found %d files' % len(names))
+    return names
+
+
+def create_otu_table(output_fp, deblurred_list):
+    """Create a biom table out of all files in a directory
 
     Parameters
     ----------
-    output_fp: string
+    output_fp : string
         filepath to final BIOM table
-    all_tables: list
-        list of filepaths for BIOM tables to merge
+    deblurred_list : list of string
+        list of file names (including path) of all deblurred
+        fasta files to add to the table
     """
-    master = load_table(all_tables[0])
-    for input_fp in all_tables[1:]:
-        master = master.merge(load_table(input_fp))
-    write_biom_table(master, output_fp)
+    logger = logging.getLogger(__name__)
+    logger.info('create_otu_table for %d samples, '
+                'into output table %s' % (len(deblurred_list), output_fp))
+
+    seqdict = {}
+    seqlist = []
+    sampset = set()
+    samplist = []
+    sizeregexp = re.compile('(?<=size=)\w+')
+    obs = scipy.sparse.dok_matrix((1E9, len(deblurred_list)), dtype=np.int)
+    for (cfilename, csampleid) in deblurred_list:
+        if csampleid in sampset:
+            warnings.warn('sample %s already in table!', UserWarning)
+            logger.error('sample %s already in table!' % csampleid)
+            continue
+        sampset.add(csampleid)
+        samplist.append(csampleid)
+        csampidx = len(sampset)-1
+        with open(cfilename, 'U') as f:
+            for chead, cseq in parse_fasta(f):
+                if cseq not in seqdict:
+                    seqdict[cseq] = len(seqlist)
+                    seqlist.append(cseq)
+                cseqidx = seqdict[cseq]
+                cfreq = float(sizeregexp.search(chead).group(0))
+                try:
+                    obs[cseqidx, csampidx] = cfreq
+                except IndexError:
+                    shape = obs.shape
+                    obs.resize((shape[0]*2,  shape[1]))
+                    obs[cseqidx, csampidx] = cfreq
+    logger.debug('loaded %d samples, %d unique sequences'
```

Comment:
It still doesn't seem to appear. An example of what I'm requesting is, for example, a blank line above this line, so it actually visually breaks the different parts of the code.